### ISSUE TITLE
Deprecate Geddy project

### ISSUE THREAD

**Issue by phanect** [geddy/geddy#745]
Unfortunately, Geddy is no-longer actively maintained.
I think it's time for the Geddy project to officially declare deprecation.

To make matters worse, there are some vulnerable versions of packages in the dependencies.
I fixed some, but I gave up to update others because it may be a big change.

Some engineers may not notice vulnerable packages included in Geddy, and use it for new projects. It is safer to notify that.

To notify Geddy is deprecated, I suggest following:

- [] Add deprecation message on the top of README
  - I will add to geddy, model, and geddy-passport projects
- [] Deprecate packages by `npm deprecate`
  - `npm deprecate geddy` "Geddy is no longer maintained."
  - `npm deprecate model` "model is no longer maintained."
  - `npm deprecate geddy-passport` "Geddy is no longer maintained."
- [] Archive the repositories under github.com/geddy
- [] Add deprecation notice on the website (geddyjs.org).
  - Or redirect geddyjs.org to github.com/geddy/geddy if @mde no longer want to maintain the website.)

If @mde agreed and no other disagreement, I will work on it except for website.

**Comment by mde** [comment]
The end of an era. :) I agree. @phanect, thank you so much for doing this. I'll take care of the Web site.

**Comment by ckhatton** [comment]
😢

If someone were to take it on, would they still be able to? I am just wondering just on the small chance I suddenly want to try maintain it.

**Comment by mde** [comment]
If someone were willing to take on the work of doing security updates, I'm happy to keep hosting the doc site, and doing releases. Let me know what y'all would like to do. I would be happy to allow the folks who want to use it to keep using it.

**Comment by myichild** [comment]
Whoops a fat finger moment 2024 ->> 2014

**Comment by ckhatton** [comment]
GitHub states it's the packages...
`parsejson`
`syntax-error`

`parsejson`: An alternative will need to be found
`syntax-error`: Just merely needs to be changed to `>1.1.1`

**Comment by ckhatton** [comment]
Right! I'm going to learn node.js. I can't see this project melt away! lol

**Comment by phanect** [comment]
@mde,

> The end of an era. :) I agree. @phanect, thank you so much for doing this. I'll take care of the Web site.

OK, I'm starting to work on.

> I would be happy to allow the folks who want to use it to keep using it.

If we archive the repositories, it would be easier for users to understand Geddy is deprecated, but by archiving repositories, entire the repository including source code, issues and pull requests would turn read-only.

Since @ckhatton might make contributions, it might be better not to archive the repositories.

@myichild and @ckhatton,
 
> Is anyone aware of any security issues I need to be mindful of as we move into production?

> Like @myichild asks, what are the security issues?

As @ckhatton already confirmed, GitHub's Vulnerability Alerts is pointing out `parsejson` and `syntax-error` as vulnerable dependencies.

In addition to these packages, `npm audit` reports more vulnerable packages: `debug`, `engine.io-client`, `ws`, and `uglify-js`.
Those packages are not direct dependencies for Geddy. If you will fix them, you need to update `socket.io-client`, `browserify`, and `jade` (now renamed as `pug`)

You can see npm's analysis with the following commands:

```sh
$ cd path/to/geddy
$ npm install                    # to generate package-lock.json
$ npm audit
```

Note that there are also vulnerabilities in `[model](https://github.com/geddy/model)`.

**Comment by myichild** [comment]
I am committed to using geddy in production until 2021 and have cloned the
app into our own repo in anticipation of its deprecation. I'm also looking
for a couple of people to help with coding on both geddy and some nodejs on
a raspberry pi reading electricity power and feeding it back to geddy.

**Comment by ckhatton** [comment]
@phanect @mde Okay, I'm not sure what to suggest. I can't say to keep the project open just on the chance of me learning nodejs enough to fix it all. It seems myself and @myichild are the only ones interested, considering there's 94 people watching this repo and no one else has said anything... yet.

**Comment by myichild** [comment]
Chris, perhaps we should have a chat?

**Comment by ckhatton** [comment]
Umm... Yeaah? haha

**Comment by myichild** [comment]
Great. Drop me a line at john@callmehelp.com.

I'm in the UK on GMT, where are you based?
---------------------------------
John Jones
UK: (+44) (0)797 644-3043

https://uk.linkedin.com/in/mrjohnjones

"To invent, you need a good imagination and a pile of junk." Edison

ᐧ

**Comment by ckhatton** [comment]
So I guess a plan of action would be myself or someone would be to:

1. Change the vulnerable packages.
2. Start working through the 100+ open issues.

@mde I remember you said in the past that you would happy to give guidance on any issues? As it is your code and you know it better than someone coming to it new.

I warn this will take ages to carry out by the way - no immediate completion. I need to learn node.js to a point I can be comfortable to understand it and use it.

**Comment by Penayo** [comment]
Hi.
An other point could be.
3. Es6 syntax support :)

**Comment by ckhatton** [comment]
@Penayo Err... haha maybe, but let's worry about the state of the project first.

**Comment by Penayo** [comment]
😕
I like Es6. Of course this is a sustancial change and it require to much
time and resources. It is not a begin point, but is an important point that
need to be in the calendar.
I can colaborate with You. Geddy is s good, I an ex rails developer and an
node.js developer, I can help you with some ideas.

**Comment by mde** [comment]
@myichild @ckhatton @Penayo Happy to add y'all as contributors if you're actually willing to do this work. How about a PR that updates for the security vulnerability first? If someone will do one of those, I'll push out a release. Otherwise it's best to to follow @phanect's suggestion.

**Comment by phanect** [comment]
@mde @myichild @ckhatton @Penayo

I have deprecated `geddy`, `model`, and `geddy-passport`.
If I misunderstand the discussion in this thread and they shouldn't be deprecated yet, I will un-deprecate it.

I also edited README and made pull requests. I was going to merge them now, but is it better If I wait for the conclusion of this discussion?

**Comment by ckhatton** [comment]
@phanect Yeaaah, it sounds like we're actually going to mend the packages. Is it possible to make it deprecated and then take the deprecated notice out? Because I am not sure how fast we'll get round to sorting out the vulnarbilities.

**Comment by ckhatton** [comment]
Going on we really need a master branch and a `develop` branch; with `feature-*`, `release-*`, and `hotfix-*` branches spanning off that.

It would be so much cleaner and organised. I can't understand most of the current branches.

https://nvie.com/posts/a-successful-git-branching-model/

**Comment by ckhatton** [comment]
> How about a PR that updates for the security vulnerability first?

@mde PR created! 😊

**Comment by ckhatton** [comment]
So with the `geddy` packages updated (#747), it is correct to say that `model` and `geddy-passport` still need updating?

**Comment by phanect** [comment]
Apologies for long delay.
I just merged updates for README to add deprecation notice. (#746, geddy/model#268, geddy/geddy-passport#30)

**Comment by phanect** [comment]
@ckhatton Sorry for ignoring this for long 🙇

> it is correct to say that model and geddy-passport still need updating?

Regarding to `model`, yes.
I have opened geddy/model#267, but I'm not sure if that's enough.
Also note that the PR doesn't update packages to the latest version, but just makes an updates to the minor version.

I don't see any vulnerable packages alert for `geddy-passport` on GitHub. I haven't confirmed with `npm audit` this time.

If you still have motivation to do this, it might be better to create another issue and discuss on it 😄

**Comment by ckhatton** [comment]
Great, thanks @phanect! I also became side tracked from fixing the Jade to Pug issue #747. Whenever I do get time, I will look into those other repos and yes I will create a new issue to discuss further on 😊
